openedx/core/lib/edx_api_utils.py

```diff
@@ -45,14 +42,7 @@ def get_edx_api_data(api_config, user, resource, resource_id=None, querystring=N
             return cached
 
     try:
-        jwt = get_id_token(user, api_config.OAUTH2_CLIENT_NAME)
```

Comment:
@rlucioni Actually this was my first thought as well and I think it would work fine for now as we have only two api clients(edxApiClient, ecoomerceClient). But the reason I decided to do this way is that in future we could have multiple clients which could lead to multiple conditional statements. So if you think that is not an issue here we can do this by using keyword argument.